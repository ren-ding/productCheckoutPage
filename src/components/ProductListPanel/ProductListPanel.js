import React, {Component} from 'react';
import PropTypes from 'prop-types';
import ProductSegment from '../ProductSegment';
import './style/ProductListPanel.css';

export default class ProductListPanel extends Component {
    render() {
        return (
            <div className='productlist-wrapper'>
                <table className="ui stripped table">
                    <thead>
                        <tr>
                        <th>Id</th>
                        <th>Name</th>
                        <th>Price</th>
                        <th></th>
                        </tr>
                    </thead>
                        <tbody>
                            {this.props.products.map(this.renderProductComponent)}
                        </tbody>
                </table>
            </div>
        );
    }

    renderProductComponent = (product, index) => (
        <ProductSegment
            key={index}
            product={product}
            addToCheckout={this.props.addToCheckout}
        />
    );
}

ProductListPanel.propTypes = {
    products: PropTypes.array.isRequired,
    addToCheckout: PropTypes.func.isRequired
};