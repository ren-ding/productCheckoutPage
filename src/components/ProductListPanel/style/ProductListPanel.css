.productlist-wrapper {
    margin-top: 15px;
}