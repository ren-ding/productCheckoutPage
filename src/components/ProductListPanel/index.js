import ProductListPanel from './ProductListPanel';
export default ProductListPanel;