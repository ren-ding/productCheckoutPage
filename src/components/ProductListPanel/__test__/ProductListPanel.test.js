import React from 'react';
import ProductListPanel from '../ProductListPanel';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
 
Enzyme.configure({ adapter: new Adapter() });

describe('ProductListPanel', ()=> {
    let productListPanel;
    let props;

    beforeEach(()=> {
        props = {
            products: [],
            addToCheckout: ()=> {}
        }

        productListPanel = () => shallow(<ProductListPanel {...props} />, { lifecycleExperimental: true });        
    });

    describe('render', () => {
       it('should render to match snapshot', () => {
           const tree = renderer.create(<ProductListPanel {...props}/>).toJSON();
           expect(tree).toMatchSnapshot();
       }); 
    });
});