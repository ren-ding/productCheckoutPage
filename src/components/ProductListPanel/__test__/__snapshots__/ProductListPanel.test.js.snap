// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListPanel render should render to match snapshot 1`] = `
<div
  className="productlist-wrapper"
>
  <table
    className="ui stripped table"
  >
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
